src/Eloquent/Model.php

```diff
@@ -77,698 +29,15 @@ abstract class Model extends BaseModel
-        }
-
-        return $dirty;
-    }
-
-    /** @inheritdoc */
-    public function originalIsEquivalent($key)
-    {
-        if (! array_key_exists($key, $this->original)) {
-            return false;
-        }
-
-        // Calling unset on an attribute marks it as "not equivalent".
-        if (isset($this->unset[$key])) {
-            return false;
-        }
-
-        $attribute = Arr::get($this->attributes, $key);
-        $original  = Arr::get($this->original, $key);
-
-        if ($attribute === $original) {
-            return true;
-        }
-
-        if ($attribute === null) {
-            return false;
-        }
-
-        if ($this->isDateAttribute($key)) {
-            $attribute = $attribute instanceof UTCDateTime ? $this->asDateTime($attribute) : $attribute;
-            $original  = $original instanceof UTCDateTime ? $this->asDateTime($original) : $original;
-
-            // Comparison on DateTimeInterface values
-            // phpcs:disable SlevomatCodingStandard.Operators.DisallowEqualOperators.DisallowedEqualOperator
-            return $attribute == $original;
-        }
-
-        if ($this->hasCast($key, static::$primitiveCastTypes)) {
-            return $this->castAttribute($key, $attribute) ===
-                $this->castAttribute($key, $original);
-        }
-
-        return is_numeric($attribute) && is_numeric($original)
-            && strcmp((string) $attribute, (string) $original) === 0;
-    }
-
-    /** @inheritdoc */
-    public function offsetUnset($offset): void
-    {
-        $offset = (string) $offset;
-
-        if (str_contains($offset, '.')) {
-            // Update the field in the subdocument
-            Arr::forget($this->attributes, $offset);
-        } else {
-            parent::offsetUnset($offset);
-
-            // Force unsetting even if the attribute is not set.
-            // End user can optimize DB calls by checking if the attribute is set before unsetting it.
-            $this->unset[$offset] = true;
-        }
-    }
-
-    /** @inheritdoc */
-    public function offsetSet($offset, $value): void
-    {
-        parent::offsetSet($offset, $value);
-
-        // Setting an attribute cancels the unset operation.
-        unset($this->unset[$offset]);
-    }
-
-    /**
-     * Remove one or more fields.
-     *
-     * @deprecated Use unset() instead.
-     *
-     * @param  string|string[] $columns
-     *
-     * @return void
-     */
-    public function drop($columns)
-    {
-        $this->unset($columns);
-    }
-
-    /**
-     * Remove one or more fields.
-     *
-     * @param  string|string[] $columns
+     * Indicates if the given model class is a MongoDB document model.
+     * It must be a subclass of {@see BaseModel} and use the
+     * {@see DocumentModel} trait.
      *
-     * @return void
+     * @param class-string|object $related
      */
-    public function unset($columns)
+    public static function isDocumentModel(string|object $related): bool
```

Comment:
This method replaces `$class instanceof MongoDB\Laravel\Eloquent\Model` to check if the model class is for a MongoDB connection. It is public and can be used in projects.